### DISCUSSION TITLE
Trying to sync two repos together on github desktop

### DISCUSSION THREAD

**Discussion by JonBYR** [community/community#36319] [Other Feature Feedback, Questions, & Ideas]
Hello,

I'm doing an assignment for university but have a bit of a dilemna that I am unsure how to solve. We need to do the following:
1.Each week, we pull from a github that contains new unit tests that we have to complete
2.After completing them, we then need to push these changes to our own repository

I'm just a bit unsure how to do this because I feel like I need to get both my own repo with the already completed unit tests and then the new repo with the new unit tests to complete and I don't exactly know how to merge these two repos together without losing any of my existing data. I have heard of something called upstream repos which I have never used before and I personally do not want to mess around with command line if I do not have to. Does anyone know how I can solve my issue?

**Comment by ossanna16** [comment]
Hey there @Memecon and thank you for asking a good question! I'm glad you found our community 🙂

**Comment by steveward** [comment]
👋🏻 @Memecon I think the best solution would be to [fork the repository](https://docs.github.com/en/get-started/quickstart/fork-a-repo) to get a copy of it to your own account. That way you can pull in the upstream changes with the new unit tests into your fork. You can clone the fork you created to GitHub Desktop, and from there you can pull in the `upstream` branches when needed. You will see `upstream/<branch_name>` in GitHub Desktop's branch list. See [this guide](https://docs.github.com/en/desktop/contributing-and-collaborating-using-github-desktop/keeping-your-local-repository-in-sync-with-github/syncing-your-branch) for how to do this in GitHub Desktop.

For the existing changes you have already made it may be easiest to just copy them over and commit them in the new fork, if possible. I hope that helps, but let me know if you have any further questions.

**Comment by airtower-luna** [comment]
Take a look at [Git Basics - Working with Remotes](https://git-scm.com/book/en/v2/Git-Basics-Working-with-Remotes), it's explained in detail there. In general I highly recommend reading at least the first 3 chapters of the Git Book. :slightly_smiling_face: 

In short: A *remote* is any other repository that you might fetch from or push to from your current, local repository. In your case there probably are two: the course repository, and your own copy on GitHub. Fetching from a remote adds its data to the database of your local repository, and then you can (for example) merge changes from a remote branch into your local branches, or create new branches from them.

"Upstream" is a conventional name for a place you get source code from that you then work with, and it's often used for the "official" version of a repository you work on, to distinguish it from your own copy. Imagine patches "flowing" down the stream to you. :wink:
